.github/workflows/build-live-branch.yml

```diff
@@ -0,0 +1,88 @@
+name: Build live branch when trunk updated
+on:
+    push:
+        branches:
+            - trunk
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
+
+permissions: {}
+
+jobs:
+    build:
+        if: github.repository_owner == 'woocommerce' && github.actor != 'github-actions[bot]'
+        runs-on: ubuntu-20.04
+        permissions:
+            contents: read
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Get current version
+              id: version
+              uses: actions/github-script@v6.3.3
+              with:
+                  script:
+                      const { getVersion } = require( './.github/workflows/scripts/get-plugin-version' );
+                      const version = await getVersion( 'woocommerce' );
+                      core.setOutput( 'version', version );
+
+            - name: Setup WooCommerce Monorepo
+              uses: ./.github/actions/setup-woocommerce-monorepo
+              with:
+                  build: false
+
+            - name: Prepare plugin zips
+              id: prepare
+              env:
+                  CURRENT_VERSION: ${{ steps.version.outputs.version }}
+              run: |
+
+                  # Current version must compare greather than any previously used current version for this PR.
+                  # Assume GH run IDs are monotonic.
+                  VERSUFFIX="${GITHUB_RUN_ID}-g$(git rev-parse --short HEAD)"
+
+                  CURRENT_VERSION="$CURRENT_VERSION-$VERSUFFIX"
+
+                  sed -i -e 's/Version: .*$/Version: '"$CURRENT_VERSION"'/' "$GITHUB_WORKSPACE/plugins/woocommerce/woocommerce.php"
+                  echo "$CURRENT_VERSION" > "$GITHUB_WORKSPACE/plugins/woocommerce/version.txt"
+
+                  cd "$GITHUB_WORKSPACE/plugins/woocommerce"
+                  bash bin/build-zip.sh
+
+                  mkdir "$GITHUB_WORKSPACE/zips"
+                  cp "$GITHUB_WORKSPACE/plugins/woocommerce/woocommerce.zip" "$GITHUB_WORKSPACE/zips/woocommerce.zip"
+                  cd "$GITHUB_WORKSPACE/zips"
+                  unzip woocommerce.zip
+                  rm woocommerce.zip
+                  mv woocommerce woocommerce-dev
+                  zip -q -r "woocommerce-dev.zip" "woocommerce-dev/"
+                  rm -fR "$GITHUB_WORKSPACE/zips/woocommerce-dev"
+
+                  # Plugin data is passed as a JSON object.
+                  PLUGIN_DATA="{}"          
+                  PLUGIN_DATA=$( jq -c --arg slug "woocommerce" --arg ver "$CURRENT_VERSION" '.[ $slug ] = { version: $ver }' <<<"$PLUGIN_DATA" )
+                  echo "plugin-data=$PLUGIN_DATA" >> $GITHUB_OUTPUT
```

Comment:
Atlas, as a follow-up task, would it be worth moving this to a dedicated script to be shared with the PR action? Just seems we might create a maintenance headache here with similar code in 2 places.

Reply:
I've got an agenda item for our team meetup at the start of June to go through all of our workflows and basically rebuild them. We have a _lot_ of redundant workflows that can be cleaned up, inconsistent job ordering, and all sorts of other things that need to be worked on. We will take a look at this one at that :smile: